syft/federated/plan.py

```diff
@@ -437,19 +490,30 @@ def request_execute_plan(
         )
         return response
 
-    def send(self, *locations):
-        """Mock send function that only specify that the Plan will have to be sent to location.
+    def send(self, *locations, force=False):
+        """Send plan to locations.
 
-        When one calls .send(), this doesn't trigger a call to remote workers, but
-        just stores "a promise" that it will be sent (with _send()) later when the plan in
-        called (and built)
+        If the plan was not built locally it will raise an exception.
```

Comment:
What is the api for building a plan? Does anything change about how it is built when it is sent to a location?

One problem I am currently running into is if the parameters for a plan are multipointers it sends the entire multipointer to a worker

Reply:
> What is the api for building a plan? 

The way I see it is that building should be dealt with internally as a consequence of a call.

This means that the user should not call `plan.build`, but the first time they execute a plan (aka `plan(<some data>`) the plan is built.

> Does anything change about how it is built when it is sent to a location?

The plan should be built locally by the worker that knowns the blueprint (the actual function/method that the plan should execute).

So the build operation shouldn't be related to sending the plan to a location.

> One problem I am currently running into is if the parameters for a plan are multipointers it sends the entire multipointer to a worker

Yes, I think the way is currently implemented that's the expected behaviour, maybe @LaRiffle knows more about it.

Reply:
Yes I think that when you call a Plan it only checks if the args provided are tensors are pointers, multipointers are seen as tensors and therefore the args are sent to the remote worker selected. `find_location` in plan.py is partially responsible for this.
However maybe this is slightly beyond the scope of this PR.

Reply:
Sounds good. That was more a question in line with my current work that I wanted to see if this addressed